emmarin/cl/ledger_proof_statements/src/tx.rs

```diff
@@ -0,0 +1,19 @@
+use cl::{
+    crust::{balance::UnitWitness, Tx, TxWitness},
+    ds::mmr::MMR,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TxPublic {
+    pub tx: Tx,
+    pub cm_mmrs: Vec<MMR>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TxPrivate {
+    pub tx: TxWitness,
+    pub mint_units: Vec<UnitWitness>,
+    pub burn_units: Vec<UnitWitness>,
+    pub spend_units: Vec<UnitWitness>,
+}
```

Comment:
do you remember why we have these duplicate wrapper structs? Should we have a single public Tx structure and a single Tx Witness?

Reply:
oh, `TxPublic` is a leftover. `TxPrivate` could also probably the remove if we move the unitwitness inside the input / mintwitness

Reply:
yes, that's how we do it in the spec, we put the unit witness with each input/mint/burn witness